decryptor/postgresql/protocol.go

```diff
@@ -174,3 +169,25 @@ func (p *PgProtocolState) HandleDatabasePacket(packet *PacketHandler) error {
 	p.lastPacketType = OtherPacket
 	return nil
 }
+
+func (p *PgProtocolState) forgetQueryState() {
+	// Query content is sensitive so we should securely remove it from memory
+	// once we're sure that it's not needed anymore.
+	if p.pendingParse != nil {
+		p.pendingParse.Zeroize()
+	}
+	p.pendingParse = nil
+
+	if p.pendingBind != nil {
+		p.pendingBind.Zeroize()
+	}
+	p.pendingBind = nil
+
+	if p.pendingExecute != nil {
+		p.pendingExecute.Zeroize()
+	}
+	p.pendingExecute = nil
+
+	// OnQuery uses "string" values and those can't be safely zeroized :(
```

Comment:
Maybe add `TODO` here? Also, we don't have to zeroize everything with `\0` byte, it can actually be any ASCII char code. This way, "zeroized" strings will remain valid even if they contained multi-byte chars. Of course, if there is a safe way of inplace string update.

Reply:
I'm afraid it's technically impossible to perform safely. In Go strings are immutable (from user standpoint). While I guess it's possible to hack into [`reflect.StringHeader`](https://golang.org/pkg/reflect/#StringHeader), I don't this it's a portable, safe, and future-proof approach.

Since zeroization is a best-effort technique, I believe it is acceptable to omit it here and rely on GC to launder the memory eventually.